lib/Conch/Controller/DeviceReport.pm

```diff
@@ -190,18 +193,18 @@ sub _record_device_report {
 
 			my $device_specs = $schema->resultset('DeviceSpec')->update_or_create(
 				{
-					device_id     => $device_id,
-					hardware_product_id    => $hw_profile->id,
-					bios_firmware => $dr->{bios_version},
-					cpu_num       => $dr->{processor}->{count},
-					cpu_type      => $dr->{processor}->{type},
-					nics_num      => $nics_num,
-					dimms_num     => $dr->{memory}->{count},
-					ram_total     => $dr->{memory}->{total},
+					device_id           => $device->id,
+					hardware_product_id => $device->hardware_product->hardware_product_profile->id,
+					bios_firmware       => $dr->{bios_version},
+					cpu_num             => $dr->{processor}->{count},
+					cpu_type            => $dr->{processor}->{type},
+					nics_num            => $nics_num,
+					dimms_num           => $dr->{memory}->{count},
+					ram_total           => $dr->{memory}->{total},
 				}
 			);
 
-			$log->info("Created Device Spec for Device $device_id");
+			$log->info("Created Device Spec for Device ".$device->id);
```

Comment:
This will not render correctly. use `$device_id` (which needs to be initialized above, in case a new device was created).

Reply:
Addressed